drivers/gyro/adxrs290/adxrs290.c

```diff
@@ -0,0 +1,324 @@
+	if (IS_ERR_VALUE(ret))
+		return ret;
+
+	reg_val &= ~ADXRS290_HPF_MASK;
+	reg_val |= (hpf << 4) & ADXRS290_HPF_MASK;
+
+	return adxrs290_reg_write(dev, ADXRS290_REG_FILTER, reg_val);
+}
+
+/**
+ * @brief Get the Gyro data channels.
+ * @param dev - Device handler.
+ * @param ch - Channel to read.
+ * @param rate - Pointer to rate value.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t adxrs290_get_rate_data(struct adxrs290_dev *dev,
+			       enum adxrs290_channel ch, int16_t *rate)
+{
+	int32_t ret = SUCCESS;
+	uint8_t data[3];
+
+	data[0] = 0x80 | (ADXRS290_REG_DATAX0+ch*2);
+	data[1] = 0x80 | (ADXRS290_REG_DATAX1+ch*2);
+	data[2] = 0;
+	ret = spi_write_and_read(dev->spi_desc, data, 3);
+	if (IS_ERR_VALUE(ret))
+		return ret;
+
+	*rate = (((int16_t)data[2]) << 8) | data[1];
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Get the temperature data.
+ * @param dev - Device handler.
+ * @param temp - Pointer to temperature value.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t adxrs290_get_temp_data(struct adxrs290_dev *dev, int16_t *temp)
+{
+	int32_t ret = SUCCESS;
+	uint8_t data[3];
+
+	data[0] = 0x80 | ADXRS290_REG_TEMP0;
+	data[1] = 0x80 | ADXRS290_REG_TEMP1;
+	data[2] = 0;
+
+	ret = spi_write_and_read(dev->spi_desc, data, 3);
+	if (IS_ERR_VALUE(ret))
+		return ret;
+
+	*temp = ((((int16_t)data[2]) << 8) | data[1]) & 0x0FFF;
+	*temp = (*temp << 4) >> 4;
+	return ret;
+}
+
+/**
+ * Initialize the device.
+ * @param device - The device structure.
+ * @param init_param - The structure that contains the device initial
+ *		       parameters.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t adxrs290_init(struct adxrs290_dev **device,
+		      const struct adxrs290_init_param *init_param)
+{
+	struct adxrs290_dev *dev;
+	int32_t ret = 0;
+	uint8_t val = 0;
+
+	dev = (struct adxrs290_dev *)malloc(sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	ret = spi_init(&dev->spi_desc, &init_param->spi_init);
+	if (IS_ERR_VALUE(ret))
+		goto error;
+
+	ret = adxrs290_reg_read(dev, ADXRS290_REG_DEV_ID, &val);
+	if (IS_ERR_VALUE(ret) || (val != ADXRS290_DEV_ID))
+		goto error;
+
+	// Enable measurement mode.
+	if (init_param->mode == ADXRS290_MODE_MEASUREMENT) {
+		ret |= adxrs290_reg_write(dev, ADXRS290_REG_POWER_CTL,
+					  ADXRS290_MEASUREMENT | ADXRS290_TSM);
+		if (IS_ERR_VALUE(ret))
+			goto error;
+
+	}
+	// Set initial Band pass filter poles.
+	ret |= adxrs290_set_lpf(dev, init_param->lpf);
+	if (IS_ERR_VALUE(ret))
+		goto error;
+
+	ret |= adxrs290_set_hpf(dev, init_param->hpf);
+	if (IS_ERR_VALUE(ret))
+		goto error;
+
+	*device = dev;
+
+	return ret;
+
+error:
+	free(dev);
+
+	return ret;
+}
+
+/**
+ * @brief Free memory allocated by adxrs290_setup().
+ * @param dev - Device handler.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t adxrs290_remove(struct adxrs290_dev *dev)
+{
+	int32_t ret;
+
+	ret = spi_remove(dev->spi_desc);
```

Comment:
nitpick: I would say to either return before `free(dev)` in case of error or not get `ret` anymore.

Reply:
I just removed the ret and just return SUCCESS at the end.